BedrockServer.h

```diff
@@ -8,30 +8,95 @@
+    // will need to pick a new master. The other is if a higher-priority master asks to stand up in our place. In this
+    // second case, we'll stand down without shutting down.
+    //
+    //
+    // # Shutting Down
+    // Let's start with shutting down. Standing down is a subset of shutting down (when we start out mastering), and
+    // we'll get to that later.
+    //
+    // When a BedroskServer comes up, it's _shutdownState is RUNNING. This is the normal operational state. When the
+    // server receives a SIGINT, that state changes to START_SHUTDOWN. This change causes a couple things to happen:
+    // 
+    // 1. The command port is closed, and no new connections are accepted from clients.
+    // 2. When we respond to commands, we add a `Connection: close` header to them, and close the socket after the
+    //    response is sent.
+    // 3. We set timeouts for any HTTPS commands to five seconds. Note that if we are slaving, this has no effect on
+    //    commands that were escalated to master, we continue waiting for those commands.
+    //
+    // The server then continues operating as normal until there are no client connections left (because we've stopped
+    // accepting them, and closed any connections as we responded to their commands), at which point it switches to
+    // CLIENTS_RESPONDED. The sync node notices this, and on it's next update() loop, it begins shutting down. If it
+    // was mastering, the first thing it does in this case is switch to STANDINGDOWN. See more on standing down in the
+    // next section.
+    //
+    // The sync node will continue STANDINGDOWN until two conditions are true:
+    // 
+    // 1. There are no commands in progress.
+    // 2. There are no outstanding queued commands.
+    //
+    // A "command in progress" is essentially any command that we have not completed but is not in our main command
+    // queue. We increment the count of these commands when we dequeue a command from the main queue, and decrememnt it
+    // when we respond to the command (and also in a few other exception cases where the command is abandoned or does
+    // not require a response). This means that if a command has been moved to the queue of outstanding HTTPS commands,
+    // or the sync thread queue, or is currently being handled by a worker, or escalated to master, it's "in progress".
+    //
+    // If we were not mastering, there is no STANDINGDOWN state to wait through - all of a slaves commands come from
+    // local clients, and once those connections are all closed, then that means every command has been responded to,
+    // implying that there are neither queued commands nor commands in progress.
+    //
+    // The sync node then finishes standing down, informing the other nodes in the cluster that it is now in the
+    // searching STATE. At this point, we switch to the DONE state. The sync thread waits for worker threads to join.
```

Comment:
i think this should be `SEARCHING state` instead of `searching STATE` to match style